lib/sycamore/sycamore/llms/openai.py

```diff
@@ -151,18 +151,23 @@ def get_guidance_model(self, model) -> Model:
             # specific assumptions about how deployed models are named that don't work
             # well with Azure. This is the only way I was able to get it to work
             # reliably.
+            # p.s. mypy seems to get mad if cls is not defined as being either, hence
+            # the union expression
             if model.is_chat:
-                cls = AzureOpenAIChat
+                cls: Union[type[AzureOpenAIChat], type[AzureOpenAICompletion]] = AzureOpenAIChat
```

Comment:
Did this fail for you in Github or locally? Did you run mypy it from the root (sycamore-monorepo) or from the lib directory? I run mypy all the time (mostly in the library directory), and it's frustrating that these errors didn't surface for me or in any of the Github commits after this change.

Reply:
I guess one other thought -- did mypy get upgraded in your poetry lock?

Reply:
(it passed local pre-commit mypy. but `poetry run mypy lib/sycamore/sycamore/llms/openai.py` failed)